File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from 'main_room'
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def send_message(data)
    b = data['body']

    m = Message.create body: b, author: current_user

    ActionCable.server.broadcast 'main_room', message: {body: m.body, author: current_user.name}
  end

end

File: app/javascript/channels/chat_channel.js
import * as $ from 'jquery';
import consumer from "./consumer"

var channel = consumer.subscriptions.create("ChatChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log('connected')
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log('receiving')
  },

  send_message(message) {
    this.perform('send_message', message)
  }
});

$(document).on('turbolinks:load', function() {
  $('#message-submit').click(function() {
    var message_body = $('#message-body').val()

    if($.trim(message_body).length > 0) {
      channel.send_message({body: message_body})
    }
  })}
)

